server/api/middleware/csrf.go

```diff
@@ -0,0 +1,19 @@
+package middleware
+
+import "net/http"
+
+func csrf(next http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		cookie, err := r.Cookie("csrf-token")
+		if err != nil {
+			http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+			return
+		}
+
+		if r.Header.Get("X-CSRF-Token") != cookie.Value {
```

Comment:
I would use https://pkg.go.dev/crypto/subtle#ConstantTimeCompare here, otherwise you might open yourself up to timing attacks.